segment/filer_test.go

```diff
@@ -22,6 +22,39 @@ func TestFileName(t *testing.T) {
 	require.Equal(t, "00000007394872394732-00ab1234cd4567ef.wal", fn)
 }
 
+func TestSegmentReadBeforeAfterFirstIndex(t *testing.T) {
+	vfs := newTestVFS()
+
+	f := NewFiler("test", vfs)
+
+	seg0 := testSegment(1)
+
+	w, err := f.Create(seg0)
+	require.NoError(t, err)
+	defer w.Close()
+
+	// Should not be able to read that from tail
+	_, err = w.GetLog(1)
+	require.Error(t, types.ErrNotFound)
+
+	// Append to writer
+	err = w.Append([]types.LogEntry{{Index: 1, Data: []byte("one")}})
```

Comment:
This test doesn't seem to be testing the case that failed without this fix - to test that we'd need to insert the first log with an index _greater_ than 1.

Did I miss something?

Reply:
Ah, I actually have that test as well, I forgot to push it.